### PR TITLE
Fix setting of texture coordinates

### LINKED ISSUE #316: [BUG] No texture coordinates when loading the sample traces.shp file from tutorial 12
When following the tutorial chapter 12 Visualizing Geological Cross Sections with PyVista no texture information is created from the command line 

mesh = gg.visualization.create_mesh_from_cross_section(linestring=traces.loc[0].geometry,
                                                       zmin=traces.loc[0]['zmin'],
                                                       zmax=traces.loc[0]['zmax'])

**To Reproduce**
Following exactly the steps of tutorial 12

This is the outcome of creating the mesh from cross sections.

>>> mesh = gg.visualization.create_mesh_from_cross_section(linestring = traces.loc[0].geometry, zmin=traces.loc[0].zmin, zmax=traces.loc[0].zmax)
>>> mesh
PolyData (0x21749143fa0)
  N Cells:    20
  N Points:   22
  N Strips:   0
  X Bounds:   3.239e+07, 3.242e+07
  Y Bounds:   5.717e+06, 5.763e+06
  Z Bounds:   -6.000e+03, 5.000e+02
  N Arrays:   0

missing the texture informations printed in the tutorials output.

Subsequently, I cannot apply the tutorials profile image as texture to the mesh.
This is the output of the line
>>> p.add_mesh(mesh, texture=tex)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\ga93sos\miniconda3\envs\Py311\Lib\site-packages\pyvista\plotting\plotter.py", line 3475, in add_mesh
    raise ValueError(
ValueError: Input mesh does not have texture coordinates to support the texture.

I have just installed the actuals gemgis versions with all actual dependencies.

Any idea ??

### PR DESCRIPTION
**Pull request by banesullivan**
# Description
This should resolve #316 and close https://github.com/pyvista/pyvista/issues/5209

# Checklist
- [ ] My code follows the [PEP 8 style guidelines](https://www.python.org/dev/peps/pep-0008/).
- [ ] My code uses type hinting for function and method arguments and return values.
- [ ] My code contains descriptive and helpful docstrings 
which are formatted per the [Google Python Style Guidelines](http://google.github.io/styleguide/pyguide.html).
- [ ] I have created tests which entirely cover my code.
- [ ] The test code either 1. demonstrates at least one valuable use case (e.g. integration tests) 
or 2. verifies that outputs are as expected for given inputs (e.g. unit tests).
- [ ] New and existing tests pass locally with my changes.

### CODE REVIEW

File: gemgis/visualization.py [hunk 1]
```diff
@@ -505,8 +505,15 @@ def create_mesh_from_cross_section(linestring: shapely.geometry.linestring.LineS
...
 
-    # Generating Tcoords
-    surface.t_coords = uv
+    # Set texture coordinates on PyVista mesh. This has been renamed a few times
+    # Reference https://github.com/pyvista/pyvista/issues/5209
+    try:
+        if pv._version.version_info < (0, 43):
+            surface.active_t_coords = uv
+        else:
+            surface.active_texture_coordinates = uv
```
Comment (banesullivan) [inline gemgis/visualization.py:514]: Maybe we want to add a third condition if the version is whenever `t_coords` was supported, but honestly that version of PyVista is not supported these days and this should use the modern API

Comment (AlexanderJuestel) [reply gemgis/visualization.py:514]: @banesullivan thanks for looking into that. I did not realize that the error was on my side during the mesh generation! I would actually keep it to the latest version of PyVista! If you don't mind just changing the old line 509 to what you have written in the new line 514 now (`surface.active_texture_coordinates = uv`), then I would be happy to merge it :)

Within the framework of the PyOpenSci Review, I will also have to extend and complete my tests.....

Comment (banesullivan) [reply gemgis/visualization.py:514]: Considering that 0.43 isn't out yet, I recommend keeping this

Comment (AlexanderJuestel) [reply gemgis/visualization.py:514]: Ah, right, should have read it more carefully. So it will be renamed in 0.43 again

Comment (banesullivan) [reply gemgis/visualization.py:514]: Definitely get this though and make sure this change is what you need. I did not run this code

Comment (AlexanderJuestel) [reply gemgis/visualization.py:514]: Just checked and your changes fix the issue! The texture is shown correctly again.

**Pull request merged**